src/components/FormAuthorization/FormAuthorization.tsx

```diff
@@ -0,0 +1,249 @@
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router-dom';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+import {
+  Avatar,
+  Box,
+  Button,
+  Container,
+  IconButton,
+  InputAdornment,
+  Link,
+  styled,
+  TextField,
+  Typography,
+} from '@mui/material';
+import {
+  AuthError,
+  createUserWithEmailAndPassword,
+  getAuth,
+  signInWithEmailAndPassword,
+} from 'firebase/auth';
+import { addDoc, collection } from 'firebase/firestore';
+import { db } from '@/Api/firebase';
+import {
+  ERROR_CODES_FIREBASE,
+  ERROR_MESSAGE,
+  REGEX_EMAIL,
+  REGEX_PASSWORD,
+  ROUTE,
+  SUCCESS_MESSAGE,
+} from '@/constants';
+import { useAppDispatch } from '@/hooks/redux';
+import { setAlertMsg } from '@/redux/slice';
+import { ILoginForm } from '@/types';
+
+const CssTextField = styled(TextField)({
+  '& label.Mui-focused': {
+    color: '#001E6A',
+  },
+  '& .MuiInput-underline:after': {
+    borderBottomColor: '#001E6A',
+  },
+  '& .MuiOutlinedInput-root': {
+    '&.Mui-focused fieldset': {
+      borderColor: '#001E6A',
+      borderLeftWidth: 6,
+      padding: '4px',
+    },
+  },
+});
+
+const FormAuthorization = ({ registration }: { registration: boolean }): JSX.Element => {
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+  const [showPassword, setShowPassword] = useState(false);
+  const dispatch = useAppDispatch();
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      email: '',
+      password: '',
+      repeatPassword: '',
+    },
+  });
+
+  const handleRegister = async (email: string, password: string): Promise<void> => {
+    try {
+      const auth = getAuth();
+      const res = await createUserWithEmailAndPassword(auth, email, password);
+      const user = res.user;
+      addDoc(collection(db, 'users'), {
+        uid: user.uid,
+        name,
+        authProvider: 'local',
+        email,
+      });
+      dispatch(setAlertMsg({ message: SUCCESS_MESSAGE(t).registration, severity: 'success' }));
+      navigate('/main');
+    } catch (error) {
+      const { code } = error as AuthError;
+
+      if (code === ERROR_CODES_FIREBASE.emailAlreadyInUse) {
+        dispatch(setAlertMsg({ message: ERROR_MESSAGE(t).emailAlreadyInUse }));
+      } else {
+        dispatch(setAlertMsg({ message: ERROR_MESSAGE(t).unknownError }));
+      }
+    }
+  };
+
+  const handleLogin = async (email: string, password: string): Promise<void> => {
```

Comment:
I think login and logout should be at least splitted, parts with request and dispatch should go to async redux actions, another part can stay in component or u can fully move it to api